### PR TITLE
Implement the zc command ##sign

### PR DESCRIPTION
**Pull request by mrmacete**
Also add zcn, zcn! to compare only signatures with the same name.

### CODE REVIEW

File: libr/anal/sign.c [hunk 1]
```diff
@@ -837,6 +837,229 @@ R_API bool r_sign_delete(RAnal *a, const char *name) {
 	return sdb_remove (a->sdb_zigns, k, 0);
 }
 
+static bool matchBytes(RSignItem *a, RSignItem *b) {
+	if (a->bytes && b->bytes) {
+		if (a->bytes->size == b->bytes->size) {
+			return !memcmp (a->bytes->bytes, b->bytes->bytes, b->bytes->size);
```
Comment (radare) [inline libr/anal/sign.c]: bytes have binary mask, that wont work for all cases, but its fine for me to merge it for now

File: libr/anal/sign.c [hunk 2]
```diff
@@ -837,6 +837,229 @@ R_API bool r_sign_delete(RAnal *a, const char *name) {
...
+		}
+	}
+	return false;
+}
+
+static bool matchGraph(RSignItem *a, RSignItem *b) {
+	if (a->graph && b->graph) {
+		if (a->graph->cc != b->graph->cc) {
```
Comment (radare) [inline libr/anal/sign.c]: i would have a counter incrementing everytime there's a match ini here, at the end you can check the % of matching ratio to return true or false

File: libr/anal/sign.c (hunk 1 above)
Comment (mrmacete) [reply libr/anal/sign.c]: yeah just copy pasta from radiff for now

Comment (mrmacete) [reply libr/anal/sign.c]: oh i see... will try to fix

File: libr/anal/sign.c [hunk 3]
```diff
@@ -837,6 +837,229 @@ R_API bool r_sign_delete(RAnal *a, const char *name) {
...
+	// do the sign diff here
+	r_list_foreach (la, itr, si) {
+		if (strstr (si->name, "imp.")) {
+			continue;
+		}
+		r_list_foreach (lb, itr2, si2) {
+			if (strstr (si2->name, "imp.")) {
+				continue;
+			}
+			if (matchBytes (si, si2)) {
```
Comment (radare) [inline libr/anal/sign.c]: i would do a matchSignatures() and call those matchBytes, matchGraph inside

File: libr/core/cmd_zign.c [hunk 4]
```diff
@@ -820,8 +829,46 @@ TODO: add useXRefs, useName
...
+	RCore *core = (RCore *) data;
+
+	switch (*input) {
+	case ' ':
+		if (!input[1]) {
+			eprintf ("usage: zc other_space\n");
+			return false;
+		}
+		return r_sign_diff (core->anal, input + 1);
+		break;
```
Comment (radare) [inline libr/core/cmd_zign.c]: unnecessary break after return

File: libr/core/cmd_zign.c [hunk 5]
```diff
@@ -820,8 +829,46 @@ TODO: add useXRefs, useName
...
+	case 'n':
+		switch (input[1]) {
+		case ' ':
+			if (!input[2]) {
+				eprintf ("usage: zcn other_space\n");
+				return false;
+			}
+			return r_sign_diff_by_name (core->anal, input + 2, false);
+			break;
```
Comment (radare) [inline libr/core/cmd_zign.c]: same

File: libr/core/cmd_zign.c [hunk 6]
```diff
@@ -820,8 +829,46 @@ TODO: add useXRefs, useName
...
+		case '!':
+			if (input[2] != ' ' || !input[3]) {
+				eprintf ("usage: zcn! other_space\n");
+				return false;
+			}
+			return r_sign_diff_by_name (core->anal, input + 3, true);
+			break;
```
Comment (radare) [inline libr/core/cmd_zign.c]: same

File: libr/core/cmd_zign.c [hunk 7]
```diff
@@ -820,8 +829,46 @@ TODO: add useXRefs, useName
...
+		default:
+			eprintf ("usage: zcn! other_space\n");
+			return false;
+		}
+		break;
+	case '?':
+		r_core_cmd_help (core, help_msg_zc);
+		break;
+	default:
+		eprintf ("usage: zc[?n!] other_space\n");
```
Comment (radare) [inline libr/core/cmd_zign.c]: $ git grep '"Usage:'|wc -l
     439
$ git grep '"usage:'|wc -l
      23
$

Comment (mrmacete) [reply libr/core/cmd_zign.c]: LOL ok, copy-pasta strikes again

**Comment by radare** [comment]
hehe then  i merge, and we can cleanup and avoid dupes later :)

**Pull request merged**